src/utils/qgis.ts

```diff
@@ -0,0 +1,322 @@
+import { create } from 'xmlbuilder2'
+import type { XMLBuilder } from 'xmlbuilder2/lib/interfaces'
+import type { OGC_SERVICE_FORMAT, OgcLayerInfo } from './ogcServices'
+import { extractBaseUrl, fetchWfsLayerNames } from './ogcServices'
+
+/**
+ * CRS (Coordinate Reference System) definitions for QGIS layers
+ * Each field corresponds to epsg.io API endpoints:
+ * - authid: EPSG code identifier (e.g. "EPSG:4326")
+ * - srid: Spatial Reference System ID - numeric part only (e.g. "4326")
+ * - proj4: Proj4 string from https://epsg.io/{code}.proj4
+ *
+ * To add a new CRS:
+ * 1. Fetch proj4 string from https://epsg.io/{code}.proj4
+ * 2. Add entry here with authid, srid, and proj4 fields
+ */
+const CRS_DEFINITIONS = {
+  'EPSG:4326': {
+    authid: 'EPSG:4326',
+    srid: '4326',
+    proj4: '+proj=longlat +datum=WGS84 +no_defs +type=crs'
+  }
+}
+
+type SupportedCrs = keyof typeof CRS_DEFINITIONS
+const DEFAULT_PROJECTION = 'EPSG:4326'
+
+type LAYER_TYPE = 'raster' | 'vector'
+
+/**
+ * Generates a unique ID for QGIS layers
+ */
+function generateId(): string {
+  return `layer_${Date.now()}_${Math.random().toString(36).slice(2, 9)}`
+}
+
+/**
+ * Adds a maplayer element with all its children
+ */
+function addMaplayer(
+  parent: XMLBuilder,
+  id: string,
+  datasource: string,
+  provider: OGC_SERVICE_FORMAT,
+  layerName: string,
+  type: LAYER_TYPE,
+  crs: SupportedCrs
+) {
+  const crsDef = CRS_DEFINITIONS[crs]
+
+  const maplayer = parent.ele('maplayer', {
+    type: type,
+    hasScaleBasedVisibilityFlag: '0'
+  })
+
+  maplayer.ele('id').txt(id)
+  maplayer.ele('datasource').txt(datasource)
+  maplayer.ele('provider', { encoding: 'UTF-8' }).txt(provider)
```

Comment:
Why the redundant encoding, and why only here?

Reply:
I'll leave it without. The defaults look sane on paper. A QGIS export does not have the attribute.

Reply:
@Thesauruv I know you like looking at entities details in WFS :-) If you see some [mojibake](https://fr.wikipedia.org/wiki/Mojibake), eg `cafÃ©` instead of `café`, tell me.